apps/main/src/llamalend/queries/validation/borrow-fields.validation.ts

```diff
@@ -34,22 +34,38 @@ export const validateRange = (range: number | null | undefined, { MaxLtv, Safe }
   })
 }
 
-export const validateMaxDebt = (debt: Decimal | undefined | null, maxDebt: Decimal | undefined | null) => {
-  test('debt', 'Debt must be less than or equal to maximum borrowable amount', () => {
-    if (debt != null && maxDebt != null) {
+export const validateMaxDebt = (
+  debt: Decimal | undefined | null,
+  maxDebt: Decimal | undefined | null,
+  isMaxDebtRequired: boolean,
+) => {
+  skipWhen(!isMaxDebtRequired, () => {
+    test('maxDebt', 'Maximum debt must be calculated before debt can be validated', () => {
+      enforce(maxDebt).isNotNullish()
+    })
+  })
+  skipWhen(maxDebt == null || debt == null, () => {
+    test('debt', 'Debt must be less than or equal to maximum borrowable amount', () => {
+      enforce(maxDebt).isNotNullish()
       enforce(debt).lte(maxDebt)
-    }
+    })
+  })
+}
+
+export const validateLeverageEnabled = (leverageEnabled: boolean | undefined | null, isLeverageRequired: boolean) => {
+  test('leverageEnabled', 'Leverage must be enabled', () => {
+    enforce(leverageEnabled).equals(true)
```

Comment:
Uh, shouldn't this equal (the now unused) `isLeverageRequired`?
EDIT: or use a `skipWhen`?